raiden/transfer/channel.py

```diff
@@ -1743,7 +1780,14 @@ def send_expired_withdraws(
             break
 
         nonce = get_next_nonce(channel_state.our_state)
-        channel_state.our_state.nonce = nonce
+        coop_settle = channel_state.our_state.initiated_coop_settle
+        if coop_settle is not None:
+            if (
+                coop_settle.total_withdraw_participant == withdraw_state.total_withdraw
+                and coop_settle.expiration == withdraw_state.expiration
```

Comment:
Is the equality operator appropriate here when checking expiration?

Reply:
Since we don't allow a coop settle and withdraws at the same time, this whole if statement should always be true, shouldn't it?

Reply:
Hmm... not sure about that, but looking at it the other way, I would say it seems ok to reset the `coop_settle` if _any_ withdraw expired. So I guess it could be simplified.

However, I don't like doing channel state changes here, this seems rather like a function that is supposed to return events, not cause side-effects like that. @ulope Any ideas for a better place?

Reply:
All these functions in here are in the end (through various layers) called from `raiden.transfer.channel.state_transition`. 
Also if you check the other functions, they all do direct state manipulation. So this is fine.

> Is the equality operator appropriate here when checking expiration?

I assume this is intended to check that the expiring withdraw is the one that belongs to the coop settle and not some other unrelated withdraw (which as @karlb said shouldn't be possible).